File: Project/UIColor-HexDemo/ViewController.m
//
//  ViewController.m
//  UIColor-HexDemo
//
//  Created by nakajijapan on 2014/02/23.
//  Copyright (c) 2014年 net.nakajijapan. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    UIButton *button = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [button setFrame:CGRectMake(110, 64, 100, 50)];
    [button setTitle:@"button" forState:UIControlStateNormal];
    [button setBackgroundColor:[UIColor colorWithHex:0xff00ff]];
    [self.view addSubview:button];

    UIButton *button2 = [UIButton buttonWithType:UIButtonTypeRoundedRect];
    [button2 setFrame:CGRectMake(110, 164, 100, 50)];
    [button2 setTitle:@"button" forState:UIControlStateNormal];
    [button2 setBackgroundColor:[UIColor colorWithHex:0xff00ff alpha:0.3]];
    [self.view addSubview:button2];

}

@end

File: Project/Demo/ViewController.swift
//
//  ViewController.swift
//  Demo
//
//  Created by nakajijapan on 5/10/15.
//  Copyright (c) 2015 net.nakajijapan. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let colors = [0xff00ff, 0x000000, 0xff0000, 0x00ff00, 0x0000ff]
        var height = 20
        
        for color:Int in colors {
            let button = UIButton(frame: CGRect(x: 50, y: height, width: 100, height: 50))
            button.setTitle("button", forState: UIControlState())
            button.backgroundColor = UIColor(hex: color)
            self.view.addSubview(button)
            height += 54
        }
        
        for color:Int in colors {
            let button = UIButton(frame: CGRect(x: 50, y: height, width: 100, height: 50))
            button.setTitle("button", forState: UIControlState())
            button.backgroundColor = UIColor(hex: color, alpha:0.3)
            self.view.addSubview(button)
            height += 54
        }
    }

}

